source/particles/particle_handler.cc

```diff
@@ -1309,16 +1333,26 @@ namespace Particles
     std::vector<unsigned int> send_offsets(n_neighbors, 0);
     std::vector<char>         send_data;
 
+    // TODO: is size cachable?
+    const unsigned int individual_particle_data_size =
+      Utilities::MPI::max(n_send_particles > 0 ?
+                            ((begin()->serialized_size_in_bytes() +
+                              (size_callback ? size_callback() : 0))) :
+                            0,
+                          parallel_triangulation->get_communicator());
```

Comment:
I don't think the `MPI_Allreduce` behind `MPI::max` is all that bad compared to the other things that happen in this function. The only thing that would concern me is the synchronization point in case different MPI ranks run very far ahead others. That would suggest to skip the TODO. The other question is under which circumstances the `individual_particle_data_size` can actually change, which indeed would suggest to use caching.

Reply:
Those are good questions. The `individual_particle_data_size` can only change if the number of properties changes. I think it is not possible right now to change the number of properties without building a new particle_handler from scratch. I think right now this should not be a significant overhead since anyway it is followed by two sets of communications with a WaitAll so we are at a section where everything will need to synchronize anyway.

Reply:
Yes, let's keep things simple.